src/misc/host_locale_macos.cpp

```diff
@@ -0,0 +1,203 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2024-2024  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#if defined(__APPLE__)
+
+#include "host_locale.h"
+
+#include "checks.h"
+
+#include <CoreFoundation/CoreFoundation.h>
```

Comment:
But how to check for the opposite condition? I can't understand why the macOS build tries to compile code within:

```C++
#if C_COREFOUNDATION
#else
(...)
#endif
```

Reply:
Oh, include dosbox.h at the very top first (I think that's the standard. It will pull in config.h that has all the defines that might be customized by the user at build time)

Reply:
OK, I understand now. I thought this was some compiler-level define, standard for modern macOS :)